tests/ui/typeck/bad-type-in-vec-push.rs

```diff
@@ -0,0 +1,20 @@
+// The error message here still is pretty confusing.
```

Comment:
We can probably check *how many* spans we have and if there's only one, with infer vars in the type, then we skip the customization? Not necessary for this PR.